docs/NativeModulesAndroid.md

```diff
@@ -89,21 +89,43 @@ ReadableArray -> Array
 ### Register the Module
 
 The last step within Java is to register the Module; this happens in the `createNativeModules` of your apps package. If a module is not registered it will not be available from JavaScript.
+In the package class it is important you override three methods: `createNativeModules`, `createJSModules` and `createViewManagers`.
+
+Note: It is necessary to have same package name of `MainActivity.java` in your packaging class and module class. Only then your module will be accessible across classes. 
```

Comment:
It is not necessary to have the same package name as the app if I understand correctly. I would suggest not to write anything about package names. Most IDEs auto add the package name when you create a new class anyways, and modules can easily be in separate packages.